### PR TITLE
Implement update test

### PR DESCRIPTION
**Pull request by VOID404**
**Description**

Changes proposed in this pull request:

- Add a test case that updates an app in compass, then verifies if it gets updated correctly on cluster

**Related issue(s)**

#15037

### CODE REVIEW

**Review by franpog859 — COMMENTED** [review]
The image is not updated. Therefore, I believe this test is not running on the pipeline yet. Let's change it and check it out! Also, please, update the PR description so we know what actually is done here in regards to the epic acceptance criteria

File: tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go [hunk 1]
```diff
@@ -49,6 +58,32 @@ func (cs *CompassRuntimeAgentSuite) TestApplication() {
...
+		return err
+	}
+
+	verify := func() bool {
+		app, err := appReader.Get(context.Background(), compassAppName, v1.GetOptions{})
+		if err != nil {
+			t.Logf("Couldn't get updated: %v", err)
+		}
+
+		return err == nil && app.Spec.Description == "Updated"
```
Comment (franpog859) [inline tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go:76]: I'd use some constant value like `UPDATED_APP_DESCRIPTION` or something like this just to show explicitly that it's a hardcoded test value. What do you think?

Comment (franpog859) [reply tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go:76]: Do we really need to check the description here though? Isn't it a job for the App Comparator and the `Compare` method executed later?

Comment (VOID404) [reply tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go:76]: We do have to somehow wait for sync. App Comparator fails the tests if the app doesn't match so it cannot be used for waiting.

**Review by franpog859 — APPROVED** [review]

**Pull request merged**